app/src/main/java/org/simple/clinic/bp/entry/BloodPressureEntrySheetControllerV2.kt

```diff
@@ -365,12 +364,35 @@ class BloodPressureEntrySheetControllerV2 @Inject constructor(
         .ofType<OpenAs.New>()
         .map { it.patientUuid }
 
-    return validDates
+    val existingBpUuidStream = openAs
+        .ofType<OpenAs.Update>()
+        .map { it.bpUuid }
+
+    val saveNewBp = validDates
         .withLatestFrom(bpChanges, patientUuidStream)
         .flatMapSingle { (date, bp, patientUuid) ->
-          val dateToInstant = date.atStartOfDay(UTC).toInstant()
-          bloodPressureRepository.saveMeasurement(patientUuid, bp.systolic, bp.diastolic, dateToInstant)
+          val dateAsInstant = date.atStartOfDay(UTC).toInstant()
+          bloodPressureRepository.saveMeasurement(patientUuid, bp.systolic, bp.diastolic, dateAsInstant)
         }
         .map { { ui: Ui -> ui.setBpSavedResultAndFinish() } }
+
+    val updateExistingBp = validDates
+        .withLatestFrom(bpChanges, existingBpUuidStream)
+        .flatMap { (date, newBp, existingBpUuid) ->
+          bloodPressureRepository.measurement(existingBpUuid)
+              .take(1)
+              .map { existingBp ->
+                val dateAsInstant = date.atStartOfDay(UTC).toInstant()
+                existingBp.copy(
+                    systolic = newBp.systolic,
+                    diastolic = newBp.diastolic,
+                    createdAt = dateAsInstant,
+                    updatedAt = dateAsInstant)
```

Comment:
There's no need to set the `updatedAt` timestamp again here since `BloodPressureRepository#updateMeasurement` does it for us.

Reply:
Fixing this